src/Environments/MacOS/ResourceFork.cs

```diff
@@ -342,7 +343,115 @@ public void Dump()
                 platform.A5World = a5world;
                 platform.A5Offset = jt.BelowA5Size;
                 segmentMap.AddSegment(a5world);
+
+                // Check for %A5Init CODE Segment and unpack global data to A5World
+                foreach ( var a5dataseg  in segmentMap.Segments.Values)
+                {
+                    if (a5dataseg.Name.Length >= 12 )
```

Comment:
These deeply nested `if`s and `do` make it hard to follow the logic. Consider switching the logic around so that if either of the two criteria doesn't fit, short circuit out of the `foreach` with a continue.